typespec-extension/src/code-model-builder.ts

```diff
@@ -1316,15 +1319,20 @@ export class CodeModelBuilder {
       this.trackSchemaUsage(schema, { serializationFormats: [KnownMediaType.Multipart] });
     }
 
-    // Implicit body parameter would have usage flag: UsageFlags.Spread, for this case we need to do body parameter flatten
-    const bodyParameterFlatten = sdkType.kind === "model" && sdkType.usage & UsageFlags.Spread && !this.isArm();
+    // Explicit body parameter @body or @bodyRoot would result to the existance of rawHttpOperation.parameters.body.property
+    // Implicit body parameter would result to rawHttpOperation.parameters.body.property be undefined
+    // see https://typespec.io/docs/libraries/http/cheat-sheet#data-types
+    const bodyParameterFlatten =
+      sdkType.kind === "model" && !rawHttpOperation.parameters.body?.property && !this.isArm();
```

Comment:
flatten or not should not be based on the model, it should be based on the client method (which is yet to be included in this migration).